frontend/components/ManageStakingPage/StakingContractSection/index.tsx

```diff
@@ -65,14 +81,28 @@ export const StakingContractSection = ({
   const stakingContractInfoForStakingProgram =
     stakingContractInfoRecord?.[stakingProgram];
 
-  const activeStakingContractInfo =
-    activeStakingProgram && stakingContractInfoRecord?.[activeStakingProgram];
-
-  const activeStakingProgramMeta =
-    activeStakingProgram && STAKING_PROGRAM_META[activeStakingProgram];
+  const activeStakingContractInfo = useMemo<
+    Partial<StakingContractInfo> | null | undefined
+  >(() => {
+    if (!isStakingContractInfoLoaded) return undefined;
+    if (activeStakingProgram === undefined) return undefined;
+    if (activeStakingProgram === null) return null;
+    return stakingContractInfoRecord?.[activeStakingProgram];
+  }, [
+    activeStakingProgram,
+    isStakingContractInfoLoaded,
+    stakingContractInfoRecord,
+  ]);
 
-  const isSelected =
-    activeStakingProgram && activeStakingProgram === stakingProgram;
+  /**
+   * Returns `true` if this stakingProgram is active,
+   * or user is unstaked and this is the default
+   */
+  const isSelectedOrUnstakedDefault = useMemo(() => {
```

Comment:
idk why, it's just hard to understand the purpose of it from the name, sorry :D 
also, for a bit better readability, maybe we can shift the names in comparisons. if we do `return stakingProgram === defaultStakingProgram;` in line 103
would be a bit easier to read `return stakingProgram === activeStakingProgram;` in 104 instead of `return activeStakingProgram === stakingProgram;."`

Reply:
yeah makes sense, will flip them